File: Extras/008 - Equalize the Array/equalizearray.rb
def equalizeArray(arr)
  sortedArray = arr.sort()
  countValues = []
  lastItem = nil
  totalCount = 0

  sortedArray.each do |item|
    if item != lastItem 
      countValues.push({
        value: item,
        repetitions: 1
      });
    else
      countValues[countValues.length - 1][:repetitions] += 1;
    end
    lastItem = item;
  end
  
  countValues = countValues.sort { |a, b| b[:repetitions] - a[:repetitions] }
  countValues.delete_at(0)

  countValues.each do |item|
    totalCount += item[:repetitions];
  end
  
  totalCount
end

p equalizeArray([2, 1, 2, 3])

File: Extras/008 - Equalize the Array/equalizearray.py
def equalizeArray(arr):
  sortedArray = sorted(arr)
  countValues = []
  lastItem = None
  totalCount = 0
  
  for item in sortedArray:
    if item != lastItem:
      countValues.append({
        "value": item,
        "repetitions": 1
      })
    else:
      countValues[len(countValues) - 1]["repetitions"] += 1
          
    lastItem = item;    
  
  countValues.sort(key = lambda item: item.get("repetitions"), reverse = True)
  countValues.pop(0)
  
  for item in countValues:
    totalCount += item["repetitions"]
  
  return totalCount


print(equalizeArray([2, 1, 2, 3]))
